Theme JSON schema: Add sticky position to settings, minHeight to styles

andrewserong opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Update `theme.json` schema to include `position.sticky` within the settings schema, and `dimensions.minHeight` within the styles schema.

Note that while `fixed` is treated as a valid value in code, it is not yet in use by any core blocks. To avoid folks thinking they can start using it by switching `fixed` to `true`, I've left it off this change for now.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Now that the sticky position support has been added to the Group block, and the feature will be in `6.2`, ensure the setting exists within the `theme.json` schema.

Also, `dimensions.minHeight` was missing from the style schema, so that's been added now.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

* Update the `theme.json` schema to reflect that `position.sticky` is a valid setting (boolean) — note that position is not valid as a _style_ within `theme.json` as it is intended to be set at the individual block level within the editor.
* Add `dimensions.minHeight` as a valid style.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

Proofread to make sure the changes look correct, and are added in the right places. I wasn't too sure how else to test it manually?

andrewserong committed:
Theme JSON schema: Add sticky position to settings

aaronrobertshaw reviewed (approved):
Thanks for bringing the schema up-to-date with the current state of position support @andrewserong 👍 

The decision to omit references to `fixed` position support makes sense.

LGTM!

andrewserong committed:
Add dimensions.minHeight to styles

aaronrobertshaw reviewed (commented):
Thanks for also adding the missing dimensions to the schema styles and settings.

It looks like this has exposed a problem with the [previous dimensions setting definition](https://github.com/WordPress/gutenberg/blob/1fc9c3a3ba65e6afa09c5b2c17a3812a286ca94a/schemas/json/theme.json#L234). It needs the object type as well I think.

andrewserong committed:
Add missing type: objects line

andrewserong commented:
Thanks for the reviews @aaronrobertshaw! Added the missing line, so 🤞 should be good now.

aaronrobertshaw reviewed (approved):
The latest tweak LGTM. Once the e2es pass, this should be good to go. 🚢

andrewserong commented:
I've added the backport label — I'm not sure if it's needed for schema updates? It was flagged in this issue (https://github.com/WordPress/gutenberg/issues/49010), though, so to be on the safe side, I've added the label.

pbiron commented:
All the references I see to the schema just point at https://raw.githubusercontent.com/WordPress/gutenberg/trunk/schemas/json/theme.json, so I don't think the schema is "distributed" with WP installations, so there's nothing that needs to be done there (since that now contains the new properties).

With regard to merging this into core, , 

I suspect that the _.md_ file will get incorporated into the the relevant handbook page during some automated sync'ing that happens after a WP release, but probably best to check with the release/documentation leads.

Mamaduka commented:
I just cherry-picked this PR to the wp/6.2 branch to get it included in the next release: 33001e5e71
